cassiopeia.go

```diff
@@ -6,18 +7,30 @@ import (
 	"regexp"
 )
 
+// I guest this is the main package of this service right?
+// so, every interface needed in this package should be declared here(e.g blogger and halloffame)
+// we also need to create interface that will be implemented by Cassiopeia
+
+// this package shuld be implemented by cassiopeia, the intereface name is my random thought so i dont realy like it.
+// every function on Cassiopeia shoud be listed here first
```

Comment:
Actually this is a good idea, never thought this one before. Can you please put some reference so I can look on to it for future use? (e.g. repo, web article, paper, book, yt video)

Reply:
https://medium.com/inside-bukalapak/how-i-design-my-code-54bfeb46d481

Reply:
please do